sympy/tensor/array/tests/test_mutable_ndim_array.py

```diff
@@ -108,6 +108,23 @@ def test_iterator():
         j += 1
 
 
+def test_getitem():
+    for ArrayType in [MutableDenseNDimArray, MutableSparseNDimArray]:
+        array = ArrayType(range(24)).reshape(2, 3, 4)
+        assert array.tolist() == [[[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]], [[12, 13, 14, 15], [16, 17, 18, 19], [20, 21, 22, 23]]]
+        assert array[0] == ArrayType([[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]])
+        assert array[0, 0].tolist() == [0, 1, 2, 3]
```

Comment:
why not array comparison?

Reply:
> If shape == (), maybe loop size should be 1?

With the existing code, the loop size will be 1 if shape == (). I was using `len(flat_list)` rather than 1 because if we allow an empty array in the future, the `._array` will be empty so that the loop size should be 0 not the default 1. In this case, `len(flat_list)` can show what the true length of array is.

> why not array comparison?
Do you want to use `ArrayType([0, 1, 2, 3])` instead of `[0, 1, 2, 3]`? I was trying to varify the test. Using array comparison is totally ok.